File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/entity/vms/VmsTerminalCommEntity.java
package com.shmedo.configlibrary.iot.cmd.entity.vms;

import android.text.TextUtils;

import com.shmedo.configlibrary.ble.interfaces.Validater;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  12/2/20 <br/>
 * 描述：     Vms 终端通信参数
 */
public class VmsTerminalCommEntity implements Validater {
    private String sn;
    private String netid;//网络号
    private String dstaddr;//通道的地址
    private String channel;//通信信道
    private String airbaud;//空中波特率

    public void setSn(String sn) {
        this.sn = sn;
    }

    public String getNetid() {
        return netid;
    }

    public void setNetid(String netid) {
        this.netid = netid;
    }

    public String getDstaddr() {
        return dstaddr;
    }

    public void setDstaddr(String dstaddr) {
        this.dstaddr = dstaddr;
    }

    public String getChannel() {
        return channel;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }

    public String getAirbaud() {
        return airbaud;
    }

    public void setAirbaud(String airbaud) {
        this.airbaud = airbaud;
    }

    @Override
    public void validate() {

    }

    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append("sn=" + sn);
        stringBuilder.append("&");

        if (!TextUtils.isEmpty(netid)) {
            stringBuilder.append("netid=" + netid);
            stringBuilder.append("&");
        }
        if (!TextUtils.isEmpty(dstaddr)) {
            stringBuilder.append("dstaddr=" + dstaddr);
            stringBuilder.append("&");
        }
        if (!TextUtils.isEmpty(channel)) {
            stringBuilder.append("channel=" + channel);
            stringBuilder.append("&");
        }
        if (!TextUtils.isEmpty(airbaud)) {
            stringBuilder.append("airbaud=" + airbaud);
            stringBuilder.append("&");
        }

        if (stringBuilder.toString().endsWith("&")) {
            stringBuilder.delete(stringBuilder.length() - 1, stringBuilder.length());
        }
        return stringBuilder.toString();
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/entity/vms/VmsTerminalCommEntity.kt
package com.shmedo.lib.device.base.iot_cmd.entity.vms

import com.shmedo.lib.device.base.Validater


/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  12/2/20 <br></br>
 * 描述：     Vms 终端通信参数
 */
class VmsTerminalCommEntity : Validater {
    var sn: String? = null
    var netid //网络号
            : String? = null
    var dstaddr //通道的地址
            : String? = null
    var channel //通信信道
            : String? = null
    var airbaud //空中波特率
            : String? = null

    override fun validate() {}

    override fun toString(): String {
        val stringBuilder = StringBuilder()
        try {
            stringBuilder.append("sn")
            stringBuilder.append("=")
            stringBuilder.append(sn)
            stringBuilder.append("&")
            for (f in javaClass.declaredFields) {
                val value = f[this]
                if (f.name != "sn" && value != null && value != "NullKey") {
                    stringBuilder.append(f.name)
                    stringBuilder.append("=")
                    stringBuilder.append(value)
                    stringBuilder.append("&")
                }
            }
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }
        if (stringBuilder.toString().endsWith("&")) {
            stringBuilder.delete(stringBuilder.length - 1, stringBuilder.length)
        }
        return stringBuilder.toString()
    }
}